MdeModulePkg/Core/Dxe/Image/Image.c

```diff
@@ -624,6 +628,15 @@ CoreLoadPeImage (
     return EFI_UNSUPPORTED;
   }
 
+  // MU_CHANGE START
+  if (!(Image->ImageContext.SupportsNx) && Image->ImageContext.ImageCodeMemoryType == EfiLoaderCode) {

```

Comment:
Perhaps move this code to a function in MemoryProtection.c to keep cohesion between memory protection related functionality.

Reply:
Moved to memoryprotection.c